Quadriga/src/main/java/edu/asu/spring/quadriga/web/config/SimpleSignInAdapter.java

```diff
@@ -0,0 +1,127 @@
+     * by Github. If the user details are present (which indicates the user is
+     * an authorized user), appropriate roles are assigned, Else if the user
+     * details are not present (which indicates the user is an unregistered /
+     * not approved user, restricted access is assigned)
+     */
+
+    @Override
+    public String signIn(String userId, Connection<?> connection, NativeWebRequest request) {
+        List<GrantedAuthority> authorities = new ArrayList<>();
+        SocialSignInStatus messageType = null;
+        IUser user = (IUser) userManager.findUserByProviderUserId(connection.getKey().getProviderUserId(),
+                connection.getKey().getProviderId());
+        // if user details is not present in the database, create a database
+        // record for the user.
+        if (user == null) {
+            user = userHelper.createUser(connection);
+            try {
+                
+                if(userManager.checkUserRequestExists(user.getUserName())){
+                    messageType = SocialSignInStatus.REGISTRATION_SUCCESS;
+                }
+                else{
+                    AccountRequest accountRequest = new AccountRequest();
+                    accountRequest.setUsername(user.getUserName());
+                    accountRequest.setName(user.getName());
+                    accountRequest.setEmail(user.getEmail());
+                    accountRequest.setSocialSignIn(true);
+                    accountRequest.setProvider(user.getProvider());
+                    accountRequest.setUserIdOfProvider(user.getUserIdOfProvider());
+            
+                    if(userManager.addNewUser(accountRequest) ){
+                        messageType = SocialSignInStatus.REGISTRATION_SUCCESS;
+                    }
+                }
+            } catch (QuadrigaStorageException | UsernameExistsException e) {
+                logger.error("Could not add user.", e);
+                messageType = SocialSignInStatus.REGISTRATION_FAILED;
+            } catch (QuadrigaNotificationException e) {
+                logger.error("Could not notify admin about the new user.", e);
+            } 
+        }
+        // if user details is present in the database, assign appropriate roles
+        // to the user.
+        else {
+
+            List<IQuadrigaRole> quadrigaRoles = user.getQuadrigaRoles();
+            if (quadrigaRoles != null) {
+                for (IQuadrigaRole quadrigaRole : quadrigaRoles) {
+                    authorities.add(new QuadrigaGrantedAuthority(quadrigaRole.getId()));
+                }
+            }
+        }
+
+        QuadrigaUserDetails userDetails = null;
+        if (messageType == null) {
+            userDetails = new QuadrigaUserDetails(user.getUserName(), user.getName(), user.getPassword(), null,
+                    user.getEmail());
+        } else {
+            return "/sociallogin?type=" + messageType.getSocialSignInStatusType();
+        }
```

Comment:
switch if/else statements. then you don't need an else block > less code and more readable